src/Raven.Server/Web/Studio/Sharding/Processors/ShardedStudioCollectionsHandlerProcessorForPreviewCollection.cs

```diff
@@ -84,12 +83,16 @@ protected override async ValueTask<long> GetTotalResultsAsync()
         return total;
     }
 
-    protected override bool NotModified(out string etag)
+    protected override async ValueTask<bool> NotModifiedAsync()
     {
-        DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Karmel, DevelopmentHelper.Severity.Normal,
-            "RavenDB-19066 Need to figure out the best way to combine ETags and send not modified. See `null` passed as etag to above new ShardedCollectionPreviewOperation()");
+        var etag = RequestHandler.GetStringFromHeaders(Constants.Headers.IfNoneMatch);
 
-        etag = null;
+        var result = await _requestHandler.ShardExecutor.ExecuteParallelForAllAsync(new ShardedCollectionPreviewOperation(RequestHandler, etag, _continuationToken));
```

Comment:
If I get it right you will do a request to all shards, just to get the not modified and then you will do the same request again in the `GetDocumentsAsync` ?

This brings me to a conclusion that `NotModifiedAsync` shouldn't be async at all..